tests/unit/models/keras/test_sampler.py

```diff
@@ -100,3 +250,73 @@ def test_ensemble_trajectory_sampler_resample_provides_new_samples_without_retra
     npt.assert_array_less(1e-4, tf.abs(evals_1 - evals_2))
     npt.assert_array_less(1e-4, tf.abs(evals_2 - evals_3))
     npt.assert_array_less(1e-4, tf.abs(evals_1 - evals_3))
+
+
+@random_seed
+def test_ensemble_trajectory_sampler_update_trajectory_updates_and_doesnt_retrace(
+    use_samples: bool,
+) -> None:
+    """
+    We do updates after updating the model, check if model is indeed changed and verify
+    that samples are new.
+    """
+    dim = 3
+    batch_size = 2
+    num_data = 100
+
+    example_data = empty_dataset([dim], [1])
+    test_data = tf.random.uniform([num_data, batch_size, dim])  # [N, B, d]
+
+    model, _, _ = trieste_deep_ensemble_model(example_data, _ENSEMBLE_SIZE)
+
+    trajectory_sampler = EnsembleTrajectorySampler(model, use_samples=use_samples)
+    trajectory = trajectory_sampler.get_trajectory()
+
+    eval_before = trajectory(test_data)
+
+    for _ in range(3):
+        x_train = tf.random.uniform([num_data, dim])  # [N, d]
+        new_dataset = Dataset(x_train, quadratic(x_train))
+        old_weights = trajectory_sampler._model.model.get_weights()
+        model.optimize(new_dataset)
+
+        trajectory_updated = trajectory_sampler.update_trajectory(trajectory)
+        eval_after = trajectory(test_data)
+
+        assert trajectory_updated is trajectory  # check update was in place
+
+        npt.assert_array_less(
+            1e-4, tf.abs(trajectory_sampler._model.model.get_weights()[0], old_weights[0])
+        )
+        npt.assert_array_less(
+            0.01, tf.reduce_max(tf.abs(eval_before - eval_after))
+        )  # two samples should be different
+
+    assert trajectory.__call__._get_tracing_count() == 1  # type: ignore
+
+
+# @pytest.mark.skip
+@random_seed
```

Comment:
this fails for sampling variant :(
it seems like sampling, even if the same seed is used, also depend on the input...

also, trajectory via sampling is not smooth (means are way smoother), despite using the same seed for each point it samples from a different part of the distribution
- in other words sampling is a no go :(
(strangely, it does work reasonably well in the HV project...)